code/WorkInProgress/nadir_antenna.dm

```diff
@@ -989,6 +989,51 @@ TYPEINFO(/obj/machinery/transception_pad)
 
 		SEND_SIGNAL(src, COMSIG_MOVABLE_POST_RADIO_PACKET, signal, INTERLINK_RANGE, freq)
 
+/obj/machinery/computer/transception/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "TransceptionInterlink")
+	ui.open()
+
+/obj/machinery/computer/transception/ui_data(mob/user)
+	. = ..()
+	var/list/pads_data = list()
+	for (var/device_index in src.known_pads)
+		pads_data += list(list(
+			"device_index" = device_index,
+			"identifier" = known_pads[device_index]["Identifier"],
+			"target_id" = known_pads[device_index]["INT_TARGETID"],
+			"location" = known_pads[device_index]["Location"],
+			"array_link" = known_pads[device_index]["Array Link"],
+		))
+	.["pads"] = pads_data
+	.["crate_count"] = length(shippingmarket.pending_crates)
+
+/obj/machinery/computer/transception/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	switch(action)
+		if ("ping")
+			src.try_pad_ping()
+		if ("receive")
```

Comment:
I'm unclear on if the act of doing these actions would update anything that could lead to a UI change directly (rather than like... queuing up a thing that would be reflected on the next 0.9s poll), but if it would then the relevant actions should do `. = TRUE`.